core/src/utils/deleter.cpp

```diff
@@ -0,0 +1,25 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "vecmem/utils/deleter.hpp"
+
+namespace vecmem::details {
+
+   deleter::deleter( std::size_t bytes, memory_resource& resource )
+   : m_bytes( bytes ), m_resource( &resource ) {
+
+   }
+
+   void deleter::operator()( void* ptr ) {
+
+      if( ptr != nullptr ) {
+         m_resource->deallocate( ptr, m_bytes );
```

Comment:
Very good point. I'll look into the `std::default_delete` and update this class accordingly.

It's actually curious that the `std::pmr` namespace doesn't include a class like this. :confused: (But I checked again, I don't see that it would...)

Reply:
Thinking about this a bit more, I think this will need some bigger changes... Which I would rather do in another PR, as this one is way too big already. :frowning: